### PR TITLE
[READY]Makes addtimer use callbacks

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Needs review and then quick merge to avoid conflict hell.

Also, keep an eye out for any pr active now or merged recently that uses addtimer, as i'll have to rebase and fix them too, and ones existing after this is merged will need to be updated.

The new addtimer syntax is addtimer(CALLBACK(thing, .proc/proctocall, args), time, TIMER_TYPE) with timertype being optional and defaulting to TIMER_NORMAL.

@Fox-McCloud this will interest you since it removed string proc references.

### CODE REVIEW

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -231,8 +231,8 @@
 	toggle_action.Grant(R)
 	return 1
 
-/obj/item/borg/uprgade/selfrepair/dropped()
-	addtimer(src, "check_dropped", 1)
+/obj/item/borg/upgrade/selfrepair/dropped()
+	addtimer(CALLBACK(src, .proc/check_dropped), 1)
```
Comment (MrStonedOne) [inline code/game/objects/items/robot/robot_upgrades.dm]: \>uprgade

**Comment by MrStonedOne** [comment]
@tgstation/commit-access help me prevent another @Cyberboss! more eyes the better =P

**Comment by Cyberboss** [comment]
Reeeeeeeee

**Comment by MrStonedOne** [comment]
<3

**Comment by Cyberboss** [comment]
I suppose you can make TIMER_NORMAL/UNIQUE not strings anymore since the main reason for that was people always forgetting that that param came before the callback arguments

**Pull request merged**